lib/Subscriber/Phpcr/RemoveSubscriber.php

```diff
@@ -55,12 +56,20 @@ public static function getSubscribedEvents()
      * remove any references to the node.
      *
      * @param RemoveEvent $event
+     *
+     * @throws DocumentReferencedException
      */
     public function handleRemove(RemoveEvent $event)
     {
         $document = $event->getDocument();
         $node = $this->documentRegistry->getNodeForDocument($document);
 
+        $references = iterator_to_array($node->getReferences());
```

Comment:
My point is that this should be done in Sulu. When somebody issues a "remove" command to the document manager it should do the minmum required to fulfil that request. Anything else is just a obligatory overhead on all requests, ignoring if they need this functionality or not.

I would rather this was done in the `sulu/sulu` library than here, otherwise we could add an option to explicitly command the DM to do this, but I would like that to be discussed more. i think it is important to keep the DM as minimal as possible.